bokeh/simpleapp.py

```diff
@@ -0,0 +1,101 @@
+import copy
+
+from .models.widgets.layouts import HBox, VBox, VBoxForm
+from .plot_object import PlotObject
+from .properties import String, Instance, Dict, Any, Either
+from .plotting import curdoc, cursession, push, show as pshow
+
+create_registry = {}
+update_registry = {}
+
+def simpleapp(*args):
+    def decorator(func):
+        wrapper = SimpleAppWrapper(func, args)
+        return wrapper
+    return decorator
+
+class SimpleAppWrapper(object):
+    def __init__(self, func, args):
+        self.name = func.__name__
+        self.widgets = list(args)
+        self.func = func
+        create_registry[func.__name__] = func
+
+    def __call__(self, *args, **kwargs):
+        return self.func(*args, **kwargs)
+
+    def update(self, func):
+        update_registry[self.name] = func
+
+    def run(self, show=True):
+        app = SimpleApp.create(self.name, self.widgets)
+        curdoc().add(app)
+        push()
+        if show:
+            pshow(app)
+        cursession().poll_document(curdoc())
+
+    def route(self, *args, **kwargs):
+        from bokeh.server.app import bokeh_app
+        from bokeh.server.utils.plugins import object_page
+        def gen():
+            app = SimpleApp.create(self.name, [x.clone() for x in self.widgets])
+            return app
+        gen.__name__ = self.name
+        bokeh_app.route(*args, **kwargs)(object_page(self.name)(gen))
+
+
+class SimpleApp(HBox):
+    name = String()
```

Comment:
docs and docstrings are coming - once we figure out what this thing is supposed to do first